python/lsst/ip/isr/isrMock.py

```diff
@@ -0,0 +1,915 @@
+            smaller center-to-edge fall off.
+
+        Notes
+        -----
+        This uses a centered 2-d Gaussian to simulate an illumination
+        pattern that falls off towards the edge of the detector.  The
+        e(x, y) coordinates are in the frame of the amplifier, and (u, v)
+        in the frame of the full trimmed image.
+        """
+        for x in range(0, ampData.getDimensions().getX()):
+            for y in range(0, ampData.getDimensions().getY()):
+                (u, v) = self.localCoordToExpCoord(amp, x, y)
+                f = np.exp(-0.5 * ((u - 100)**2 + (v - 100)**2) / size**2)
+                ampData.getArray()[y][x] = (ampData.getArray()[y][x] * f)
+
+
+class RawMock(IsrMock):
+    r"""Generate a raw exposure suitable for ISR.
+    """
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.config.isTrimmed = False
+        self.config.doGenerateImage = True
+        self.config.doGenerateDict = False
+        self.config.doOverscan = True
+        self.config.doSky = True
+        self.config.doSource = True
+        self.config.doCrosstalk = False
+        self.config.doBias = True
+        self.config.doDark = True
+
+
+class TrimmedRawMock(RawMock):
+    r"""Generate a trimmed raw exposure.
+    """
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.config.isTrimmed = True
+        self.config.doOverscan = False
+
+
+class CalibratedRawMock(RawMock):
+    r"""Generate a trimmed raw exposure.
+    """
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.config.isTrimmed = True
+        self.config.doOverscan = False
+        self.config.doBias = True
+        self.config.doDark = False
+        self.config.doFlat = False
+        self.config.doFringe = False
+        self.config.doCrosstalk = True
+        self.config.biasLevel = 0.0
+        self.config.biasSigma = 0.0
+        self.config.skySigma = 100.0
+
+
+class RawDictMock(RawMock):
+    r"""Generate a raw exposure dict suitable for ISR.
+    """
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.config.doGenerateDict = True
+
+
+class MasterMock(IsrMock):
+    r"""Parent class for those that make master calibrations.
+    """
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.config.isTrimmed = True
+        self.config.doGenerateImage = True
+        self.config.doOverscan = False
+        self.config.doSky = False
+        self.config.doSource = False
+        self.config.doCrosstalk = False
+
+        self.config.doBias = False
+        self.config.doDark = False
+        self.config.doFlat = False
+        self.config.doFringe = False
+
+
+class BiasMock(MasterMock):
+    r"""Simulated master bias calibration.
+    """
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.config.doBias = True
+        self.config.biasSigma = 30.0
```

Comment:
Everything is relative, but this still seems like a very large number for a master calib. LSST sensors have a readnoise under 10e, and that's without beating that down with numbers, as with masters.